File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/TransactionIntent.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.TransactionIntentHash
import com.radixdlt.sargon.CompiledTransactionIntent
import com.radixdlt.sargon.TransactionIntent
import com.radixdlt.sargon.transactionIntentCompile
import com.radixdlt.sargon.transactionIntentHash

fun TransactionIntent.hash(): TransactionIntentHash = transactionIntentHash(intent = this)

fun TransactionIntent.compile(): CompiledTransactionIntent = transactionIntentCompile(intent = this)

File: crates/sargon-uniffi/src/wrapped_radix_engine_toolkit/low_level/transaction_intent.rs
use crate::prelude::*;
use sargon::TransactionIntent as InternalTransactionIntent;

#[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Record)]
pub struct TransactionIntent {
    pub header: TransactionHeader,
    pub manifest: TransactionManifest,
    pub message: Message,
}

impl From<InternalTransactionIntent> for TransactionIntent {
    fn from(value: InternalTransactionIntent) -> Self {
        Self {
            header: value.header.into(),
            manifest: value.manifest.into(),
            message: value.message.into(),
        }
    }
}

impl Into<InternalTransactionIntent> for TransactionIntent {
    fn into(self) -> InternalTransactionIntent {
        InternalTransactionIntent {
            header: self.header.into(),
            manifest: self.manifest.into(),
            message: self.message.into(),
        }
    }
}

#[uniffi::export]
pub fn new_transaction_intent_sample() -> TransactionIntent {
    InternalTransactionIntent::sample().into()
}

#[uniffi::export]
pub fn new_transaction_intent_sample_other() -> TransactionIntent {
    InternalTransactionIntent::sample_other().into()
}

#[uniffi::export]
pub fn transaction_intent_hash(intent: &TransactionIntent) -> IntentHash {
    intent.into_internal().intent_hash().into()
}

#[uniffi::export]
pub fn transaction_intent_compile(intent: &TransactionIntent) -> BagOfBytes {
    intent.into_internal().compile().into()
}
